### PR TITLE
added boyer moore algo in java

### LINKED ISSUE #1666: Implementation of boyer moore algorithm in java

### PR DESCRIPTION
**Pull request by sonali623**
Fixes #1666

#### Checklist:
- [ ] 4 space indentation.
- [x] Coding conventions are followed.
- [ ] Input is taken dynamically.
- [ ] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- Boyer moore algorithmn added
- 

#### Languages Used:
- Java
- 

#### Files Added:
- Boyer_Moore.java
- 

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

File: Boyer_Moore_Algorithm/Boyer_Moore.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+		else
+			s += max(1, j - badchar[txt[s+j]]); 
+	} 
+	} 
+
+	/* Driver program*/
+	public static void main(String []args) { 
+		
+		char txt[] = "ABAAABCD".toCharArray(); 
+		char pat[] = "ABC".toCharArray();
```
Comment (MastersAbh) [inline Boyer_Moore_Algorithm/Boyer_Moore.java]: Please take dynamic inputs

File: Boyer_Moore_Algorithm/Boyer_Moore.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
+/* Author : Sonali Patel*/
+/* Java Program for Boyer Moore String Matching Algorithm */
+
+
+class BoyerMoore{ 
+	
+	static int NO_OF_CHARS = 256;
```
Comment (MastersAbh) [inline Boyer_Moore_Algorithm/Boyer_Moore.java]: Give 4 space indentation across the file

File: Boyer_Moore_Algorithm/Boyer_Moore.java [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+	/* Driver program*/
+	public static void main(String []args) { 
+		
+		char txt[] = "ABAAABCD".toCharArray(); 
+		char pat[] = "ABC".toCharArray(); 
+		search(txt, pat); 
+	} 
+} 
+
```
Comment (MastersAbh) [inline Boyer_Moore_Algorithm/Boyer_Moore.java]: Add a sample input/output below the file

**Comment by sonali623** [comment]
I will work on the changes requested.

**Comment by sonali623** [comment]
@MastersAbh I have made the changes as requested.

File: Boyer_Moore_Algorithm/Boyer_Moore.java [hunk 4]
```diff
@@ -0,0 +1,91 @@
+/* Author : Sonali Patel*/
+/* Java Program for Boyer Moore String Matching Algorithm */
+import java.util.*;
+
+class Main{ 
+	
+    static int NO_OF_CHARACTERS = 256; 
+	
+	static int max_of_a_and_b (int a, int b) { return (a > b)? a: b; }
```
Comment (MastersAbh) [inline Boyer_Moore_Algorithm/Boyer_Moore.java]: I see 6 spaces here, please follow 4 space indentation across the file

File: Boyer_Moore_Algorithm/Boyer_Moore.java [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+
+class Main{ 
+	
+    static int NO_OF_CHARACTERS = 256; 
+	
+	static int max_of_a_and_b (int a, int b) { return (a > b)? a: b; } 
+
+	static void badCharHeuristic( char []str, int size,int badcharacter[]) { 
+	int i; 
+
```
Comment (MastersAbh) [inline Boyer_Moore_Algorithm/Boyer_Moore.java]: Indentation for this function

File: Boyer_Moore_Algorithm/Boyer_Moore.java [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+	// Initialize all elements of badcharacter as -1 
+	for (i = 0; i < NO_OF_CHARACTERS; i++) 
+		badcharacter[i] = -1; 
+
+	// Fill the actual value of last occurrence 
+	// of a character 
+	for (i = 0; i < size; i++) 
+		badcharacter[(int) str[i]] = i;
```
Comment (MastersAbh) [inline Boyer_Moore_Algorithm/Boyer_Moore.java]: Dont use TAB for indentation, use 4 spaces instead

File: Boyer_Moore_Algorithm/Boyer_Moore.java [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+	// Fill the actual value of last occurrence 
+	// of a character 
+	for (i = 0; i < size; i++) 
+		badcharacter[(int) str[i]] = i; 
+	} 
+
+	// A pattern searching function 
+	static void search( char text[], char pattern[]) 
+	{ 
+	int m = pattern.length;
```
Comment (MastersAbh) [inline Boyer_Moore_Algorithm/Boyer_Moore.java]: Indentation

**Comment by sonali623** [comment]
@MastersAbh I have fixed the indentation. If there's anything else please let me know.

**Comment by raghav-dalmia** [comment]
Looks good to me.

**Pull request merged**